SingularityBase/pom.xml

```diff
@@ -67,6 +67,57 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-shade-plugin</artifactId>
+          <configuration combine.self="override">
+            <minimizeJar>true</minimizeJar>
+            <shadedArtifactAttached>true</shadedArtifactAttached>
+            <shadedClassifierName>shaded</shadedClassifierName>
+            <artifactSet>
+              <includes>
+                <include>com.hubspot:SingularityBase</include>
+                <include>com.google.guava:guava</include>
+              </includes>
+            </artifactSet>
+            <filters>
+              <filter>
+                <artifact>com.google.guava:guava</artifact>
+                <excludes>
+                  <exclude>com/google/common/base/Optional.class</exclude>
+                </excludes>
+              </filter>
+            </filters>
+            <relocations>
+              <relocation>
+                <pattern>org.apache.mesos</pattern>
+                <shadedPattern>com.hubspot.singularity.shaded.org.apache.mesos.oneonetwo</shadedPattern>
```

Comment:
I don't think think that this will solve your problem. The issue isn't using an unshaded mesos, it's that mesos is using an unshaded protobuf that conflicts with what we want. So you want to rewrite the protobuf classes inside mesos, not mesos itself.